packages/grid/x-data-grid-pro/src/hooks/features/columnReorder/useGridColumnReorder.tsx

```diff
@@ -98,6 +102,8 @@ export const useGridColumnReorder = (
       dragColNode.current.classList.add(classes.columnHeaderDragging);
       if (event.dataTransfer) {
         event.dataTransfer.effectAllowed = 'move';
+        // On Android, if event.dataTransfer.setData is not called it doesn't allow to drag
+        event.dataTransfer.setData('text/plain', params.field);
```

Comment:
```suggestion
        event.dataTransfer.setData('text/html', params.field);
```
This makes it work on iOS as well 🎉

Before:

https://github.com/mui/mui-x/assets/13808724/3c300be7-b988-422b-baa7-c2a200144e4a

After:

https://github.com/mui/mui-x/assets/13808724/c8b188e8-9e71-4fb9-925b-3c65e9eca45e

It's not perfect (you can see the draggable image is shifted from 00:21), but it's definitely usable